recipe/install-clang.sh

```diff
@@ -6,7 +6,9 @@ CHOST=${macos_machine}
 
 pushd "${PREFIX}"/bin
   ln -s clang ${CHOST}-clang
-  if [[ "${CBUILD}" != ${CHOST} ]]; then
+  if [[ "${CBUILD}" != ${CHOST} ]] && [[ "${target_platform}" != linux-* ]]; then
+    # on linux, the `clang` package already has a $TRIPLE-clang, see
+    # https://github.com/conda-forge/clangdev-feedstock/pull/251
     ln -s clang ${CBUILD}-clang
```

Comment:
@isuruf

This adaptation is necessary for clang 16 & 17 post https://github.com/conda-forge/clangdev-feedstock/pull/251 & 252, but fails for clang 15. It's also not consistent between platforms. AFAICT we have a few options:
* drop clang 15 here
* backport the sysroot-patch to clang 15
* make this branch version-dependent
* package `bin/${CBUILD}-clang` into `clang` also on osx, and remove this branch entirely

Any preferences?

Reply:
@isuruf, could you please let me know your preferences for dealing with the new symlinks resp. the current differences between platforms & versions?

Reply:
Let's make the branch version dependent. That seems like the easiest.

Reply:
Fine by me. Long-term, what do you think about having the same symlink structure (across outputs) between linux and osx?

Reply:
Yeah, we should have the same structure if possible.

Reply:
OK. So if we want to unify this, currently we have:
| output | linux-* | osx-* |
|-|-|-|
| `gcc_impl_`<br/>`$target_platform` | `bin/x86_64-conda-linux-gnu-cc...`<br/>`bin/x86_64-conda_cos6-linux-gnu-cc...`<br/>`lib/gcc/x86_64-conda-linux-gnu/13.2.0/include` | - |
| `gcc_`<br/>`$target_platform` | `.../activate-gcc_linux-64.sh`<br/>`.../deactivate-gcc_linux-64.sh` | - |
| `gcc` | `bin/gcc` | - |
| `clang_impl_`<br/>`$target_platform` | - | `bin/x86_64-apple-darwin13.4.0-clang`<br/>or<br/>`bin/arm64-apple-darwin20.0.0-clang` |
| `clang_`<br/>`$target_platform` | - | `.../activate_clang_osx-64.sh`<br/>`.../deactivate_clang_osx-64.sh` |
| `clang` | `bin/clang`</br>`bin/x86_64-conda-linux-gnu-clang` | `bin/clang` |


Perhaps the cleaner approach would be to create an activation feedstock for clang on linux (or do it in this one?) and then have `bin/x86_64-conda-linux-gnu-clang` be under `clang_impl_linux-64`?

Reply:
Ping @isuruf. From the above, I'm guessing we could merge this as-is, and revisit the subject once we have `clang_linux-*` builds.

Reply:
> I'm guessing we could merge this as-is, and revisit the subject once we have `clang_linux-*` builds.

I opened #117 and #118